ipmipower/ipmipower_connection.c

```diff
@@ -335,29 +332,74 @@ _connection_setup (struct ipmipower_connection *ic, const char *hostname)
   strncpy (ic->hostname, hostname_second_parse_ptr, MAXHOSTNAMELEN);
   ic->hostname[MAXHOSTNAMELEN] = '\0';
 
-  /* Determine the destination address */
-  bzero (&(ic->destaddr), sizeof (struct sockaddr_in));
-  ic->destaddr.sin_family = AF_INET;
-  ic->destaddr.sin_port = htons (port);
-
-  if (!(result = gethostbyname (ic->hostname)))
+  snprintf(port_str, sizeof (port_str), "%d", port);
+  memset(&ai_hints, 0, sizeof (struct addrinfo));
+  ai_hints.ai_family = AF_UNSPEC;
+  ai_hints.ai_socktype = SOCK_DGRAM;
+  ai_hints.ai_flags = (AI_V4MAPPED | AI_ADDRCONFIG);
+  if ((result = getaddrinfo (ic->hostname, port_str, &ai_hints, &ai_res )) != 0)
     {
-      if (h_errno == HOST_NOT_FOUND)
+      if (result == EAI_NODATA)
```

Comment:
result is of type "hostent", so there's an issue here and to the gai_strerror call bellow.